src/main/java/com/hida/DatabaseManager.java

```diff
@@ -0,0 +1,261 @@
+
+                // a statement that allows database/webservice communication
+                Statement insertStatement
+                        = DATABASE_CONNECTION.createStatement();
+
+                // execute query
+                insertStatement.executeUpdate(sqlQuery);
+                System.out.println("id added");
+
+                // clean-up                
+                insertStatement.close();
+            }
+
+        } catch (SQLException exception) {
+            System.err.println(exception.getMessage() + " in addId");
+            System.err.println(Arrays.toString(exception.getStackTrace()));
+        }
+        System.out.println("printed to database");
+        // prints out to server what ids were created
+        //printData();
+        
+    }
+
+    /**
+     * Checks a list of ids against a database to ensure that no redundancies
+     * are added.
+     *
+     * @param list - list of ids to check.
+     * @return - a list containing ids that already exist in database. Returns
+     * an empty list of all the ids given in the param list are unique.
+     */
+    public Set<String> checkId(Set<String> list) {
+        // a list containing redundant ids
+        Set<String> redundantIdList = new HashSet();
+        try {
+
+            // a string to query a specific id
+            String sqlQuery;
+
+            for (String id : list) {
+
+                // a string to query a specific id for existence in database
+                sqlQuery = String.format("SELECT %s FROM %2$s WHERE "
+                        + "%1$s = '%3$s'", COLUMN_NAME, TABLE_NAME, id);
+
+                // a statement that allows database/webservice communication
+                Statement databaseStatement
+                        = DATABASE_CONNECTION.createStatement();
+
+                // execute statement and retrieves the result
+                ResultSet databaseResponse
+                        = databaseStatement.executeQuery(sqlQuery);
+
+                // adds id to redundantIdList if it already exists in database
+                if (databaseResponse.next()) {
+                    System.out.print("redundancy list: " + id);
+                    redundantIdList.add(id);
+                }
+
+                // clean-up                
+                databaseResponse.close();
+                databaseStatement.close();
+
+            }
+
+        } catch (SQLException exception) {
+            System.err.println(exception.getMessage() + " in checkId");
+            System.err.println(Arrays.toString(exception.getStackTrace()));
+        }
+        return redundantIdList;
+    }
+
+    /**
+     * Using a given prefix and length, this method will count and return 
+     * the number of matching ids in the database.
+     * 
+     * @param prefix - given prefix
+     * @param length - length of each id
+     * @return - returns the number of matches
+     */
+    public int checkPrefix(String prefix, int length) {

```

Comment:
This is very inefficient.  I think you can improve this greatly by adding a table that keeps track of how many IDs for a (prefix, length) pair are inserted in the database; then this can just be a simple lookup method